### PR TITLE
Replace np.NaN with np.nan in preparation for numpy 2.0

### PR DESCRIPTION
**Pull request by TomNicholas**
See https://numpy.org/devdocs/numpy_2_0_migration_guide.html#main-namespace

- [ ] ~~Closes #xxxx~~
- [ ] ~~Tests added~~
- [ ] ~~Changes are documented in `docs/releases.rst`~~
- [ ] ~~New functions/methods are listed in `api.rst`~~

### CODE REVIEW

**Comment by TomNicholas** [comment]
@maxrjones the test failure here looks weird - is something going wrong with the workflows you set up earlier?

**Comment by maxrjones** [comment]
> @maxrjones the test failure here looks weird - is something going wrong with the workflows you set up earlier?

ah sorry about that. in order to test distributions on non-tagged commits with TestPyPI, we need to use 'no-local-version' (e.g., 0.1.dev159) rather than the default of 'guess-next-dev', but also there cannot be multiple releases with the same name. I used up a bunch of the devN names on TestPyPI (see https://test.pypi.org/manage/project/virtualizarr/releases/) when testing out the workflow, so the error is reporting that name conflict. This won't be a problem if we only run the workflow on releases (configured in https://github.com/zarr-developers/VirtualiZarr/pull/140), which is how xarray has their release workflow configured.

In summary, you can ignore the issue and merge in https://github.com/zarr-developers/VirtualiZarr/pull/140 so it doesn't show up if you want.
